core/src/main/java/org/infinispan/configuration/cache/HashConfiguration.java

```diff
@@ -138,6 +148,10 @@ public float capacityFactor() {
       return capacityFactor.get();
    }
 
+   public KeyPartitionerFactory keyPartitionerFactory() {
+      return keyPartitionerFactory.get();
+   }
+
```

Comment:
Can this config be mutable? Main reason is to set a custom partitioner in a ```cacheStarting``` method of a ```LifecycleManager```